core/templates/services/platform-feature.service.ts

```diff
@@ -286,14 +290,17 @@ export class PlatformFeatureService {
     const cookieStrs = this.windowRef.nativeWindow.document.cookie.split('; ');
     const cookieMap = new Map(
       cookieStrs.map(cookieStr => <[string, string]>cookieStr.split('=')));
-
-    if (cookieMap.has(PlatformFeatureService.COOKIE_NAME_FOR_SESSION_ID)) {
-      return cookieMap.get(PlatformFeatureService.COOKIE_NAME_FOR_SESSION_ID);
+    const sessionId = (
+      cookieMap.get(PlatformFeatureService.COOKIE_NAME_FOR_SESSION_ID)
+    );
+    if (sessionId !== undefined) {
+      return sessionId;
     }
-    if (cookieMap.has(
-      PlatformFeatureService.COOKIE_NAME_FOR_SESSION_ID_IN_DEV)) {
-      return cookieMap.get(
-        PlatformFeatureService.COOKIE_NAME_FOR_SESSION_ID_IN_DEV);
+    const sessionIdInDev = (
+      cookieMap.get(PlatformFeatureService.COOKIE_NAME_FOR_SESSION_ID_IN_DEV)
+    );
+    if (sessionIdInDev !== undefined) {
+      return sessionIdInDev;
```

Comment:
Should an error be thrown if `sessionId` and `sessionIdInDev` are undefined?

Reply:
No, there is no need since we are returning `null` if both `sessionId` and `sessionIdInDev` are `undefined`